### PR TITLE
Adding Spark error messages for Databricks provider

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description
Making sure that Databricks returns the error of each run whenever it fails. 

### Select type(s) of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have fixed any merge conflicts

### CODE REVIEW

File: provider/spark.go [hunk 1]
```diff
@@ -478,12 +479,47 @@ func (db *DatabricksExecutor) RunSparkJob(args []string, store SparkFileStore) e
...
+	runID := runs[0].RunId
+	request := jobs.GetRunRequest{
+		RunId: runID,
+	}
+
+	// in order to get the status of the run output, we need to
+	// use API v2.0 instead of v2.1. The following code leverages
+	// version 2.0 of the API to get the run output.
+	var runOutput jobs.RunOutput
+	path := "/api/2.0/jobs/runs/get-output"
```
Comment (epps) [inline provider/spark.go:514]: Couple of questions here:
1. Why is this? Or I should ask, will the "why" eventually change, and if so, will we have to update `path`?
2. Do we want to move the value of `path` into configuration with a constant default? Is this overkill?

**Review by epps — APPROVED** [review]
I left a couple of questions that you may choose to ignore, but otherwise, this looks good to me.

File: provider/spark.go [hunk 2]
```diff
@@ -351,9 +354,10 @@ const (
 )
 
 type DatabricksExecutor struct {
-	client  *databricks.WorkspaceClient
-	cluster string
-	config  pc.DatabricksConfig
+	client    *databricks.WorkspaceClient
+	cluster   string
+	config    pc.DatabricksConfig
+	runClient *dbClient.DatabricksClient
```
Comment (aolfat) [inline provider/spark.go:360]: what is the diff between client & runclient?

Comment (aolfat) [reply provider/spark.go:360]: A few q's
- Is the run clients primary purpose to fetch logs from dbrix?
- Is there a case where the regular client can still connect but the runClient can fail? My concern is, if the answer to above is yes (runClient purely for log fetching), if we run on dbricks and there are no failures but there is a problem with connecting to the run client our entire dbricks executor fails.

File: provider/spark.go [hunk 3]
```diff
@@ -478,12 +479,47 @@ func (db *DatabricksExecutor) RunSparkJob(args []string, store SparkFileStore) e
...
 }
 
+func (db *DatabricksExecutor) getErrorMessage(jobId int64) (string, error) {
+	ctx := context.Background()
+
+	runRequest := jobs.ListRunsRequest{
+		JobId: jobId,
+	}
+
+	runs, err := db.client.Jobs.ListRunsAll(ctx, runRequest)
```
Comment (aolfat) [inline provider/spark.go:500]: is there a way to only fetch with runrequest?

File: provider/spark.go [hunk 4]
```diff
@@ -430,29 +434,26 @@ func NewDatabricksExecutor(databricksConfig pc.DatabricksConfig) (SparkExecutor,
 			Username: databricksConfig.Username,
 			Password: databricksConfig.Password,
 		}))
+
+	runClient, err := dbClient.New(&dbConfig.Config{
```
Comment (aolfat) [inline provider/spark.go:438]: should we only create this when we fetch errors? From my naive understanding I imagine the flow should be like this.

- We run on dbrix
- Something fails (we return the error message that we get from fmt.error)
- We attempt to get the more verbose message from dbrix but if it fails its fine and we can still default to the other error?

**Review by aolfat — APPROVED** [review]

**Pull request merged**